### PR TITLE
Feature/relay control

### PR DESCRIPTION
**Pull request by Zer0-bit**
Adds a couple new things:
1. Relay control for the brew button and solenoid valve
2. Stop shot on weight reached
3. UI fixes and some predefined ratios to select from:
image

### CODE REVIEW

File: src/gaggiuino.ino [hunk 1]
```diff
@@ -519,7 +537,10 @@ void trigger1(void) {
       eepromCurrentValues.warmupState       = myNex.readNumber("warmupState");
       break;
     case 5:
-      break;
+      eepromCurrentValues.stopOnWeightState = myNex.readNumber("shotState");
+      eepromCurrentValues.shotDose = myNex.readNumber("shotDose") / 10.f;
+      eepromCurrentValues.shotPreset = myNex.readNumber("shotPreset");
+      eepromCurrentValues.shotStopOnCustomWeight = myNex.readNumber("shotCustomVal") / 10.f;
```
[the remark this answers is not in the document]
Comment (Zer0-bit) [reply src/gaggiuino.ino:543]: perhaps 🤣

File: src/gaggiuino.ino [hunk 2]
```diff
@@ -720,19 +742,29 @@ static void manualPressureProfile(void) {
...
+
+  if ( brewState() && !stopOnWeight()) {
+    if(!paramsReset) {
+      brewParamsReset();
+      paramsReset = true;
     }
+    openValve();
     brewActive = true;
-  } else{
+  } else if ( brewState() && stopOnWeight()) {
```
Comment (Havald) [inline src/gaggiuino.ino:754]: stopOnWeight() is implicitly true here if brewState() is checked again, on the next else branch !brewState() is then implicit, so consider removing them for efficiency

Comment (Zer0-bit) [reply src/gaggiuino.ino:754]: Yeah it was there for visibility when i was testing, i'm changing the logic anyway lol

**Review by Havald — APPROVED** [review]

**Comment by Zer0-bit** [comment]
Deprecated by the PR relay control 2

**Pull request closed without merging**